style_guide.md

```diff
@@ -0,0 +1,307 @@
+          when: [rules.isNotProxy, rules.isEmployed],
+        },
+        {
+          question: question(isProxy=true, isEmployed=true),
+          when: [rules.isProxy, rules.isEmployed],
+        },
+        {
+          question: question(isProxy=false, isEmployed=false),
+          when: [rules.isNotProxy, rules.isNotEmployed],
+        },
+        {
+          question: question(isProxy=true, isEmployed=false),
+          when: [rules.isProxy, rules.isNotEmployed],
+        },
+      ],
+    };
+  
+    local question(isProxy, isEmployed) = {
+      id: 'business-name-question',
+      title: questionTitle(isProxy, isEmployed),
+      description: description,
+      type: 'MutuallyExclusive',
+      mandatory: false,
+      answers: [
+        {
+          id: 'business-name-answer',
+          label: 'Organisation or business name',
+          max_length: 160,
+          mandatory: false,
+          type: 'TextField',
+        },
+        {
+          id: 'no-business-name-answer',
+          mandatory: false,
+          type: 'Checkbox',
+          options: [
+            {
+              label: option,
+              value: option,
+            },
+          ],
+        },
+      ],
+    };
+  
+    local questionTitle(isProxy, isEmployed) = (
+      if isEmployed then (
+        if isProxy then 
+          {
+            text: 'What is the main activity of <em>{person_name_possessive}</em> organisation, business or freelance work?',
+            placeholders: [
+              placeholders.personNamePossessive,
+            ]
+          }
+        else 'What is the main activity of your organisation, business or freelance work?'
+      ) else (
+        if isProxy then 
+          {
+            text: 'What was the main activity of <em>{person_name_possessive}</em> organisation, business or freelance work?',
+            placeholders: [
+              placeholders.personNamePossessive,
+            ]
+          }
+        else 'What was the main activity of your organisation, business or freelance work?';
+      )
+    );
+    ```
+- The `isProxy` and `isEmployed` variables are passed to the `question` method.
+- The `question` method calls `questionTitle(isProxy, isEmployed)` to resolve the question title.
+- `questionTitle` returns the appropriate JSON for title.
+
+- Same rules apply to other blocks, e.g. for guidance with a `questionGuidance` method we would use:
+    ```
+    local guidance(isProxy) = (
+      if std.extVar('region_code') == 'GB-WLS' then
+        if isProxy then
+          walesGuidanceProxy else walesGuidanceNonProxy
+      else if isProxy then englandGuidanceProxy else englandGuidanceNonProxy
+    );
+    ```
+### Routing
+
+- Default routing rule is used in every block, even if the succeeding block is the target. It makes the rules easier to understand:
+    ```
+    routing_rules: [
+      {
+        goto: {
+          block: 'main-employment-block',
+        },
+      },
+    ],
+    ```
```

Comment:
I don't think the example adds anything as the block could be any block in the survey. ```suggestion
- Every block must have a routing rule, even if the succeeding block is the target. This makes the rules easier to understand.
```